database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -20,25 +20,26 @@
 import com.google.firebase.database.Query;
 
 import java.util.ArrayList;
-import java.util.List;
+import java.util.HashMap;
 
 /**
  * This class implements an array-like collection on top of a Firebase location.
  */
 class FirebaseArray implements ChildEventListener {
-    public interface OnChangedListener {
+    interface OnChangedListener {
```

Comment:
How will anyone subclass OnChangedListener if it's inside a package private class?  They won't be able to see it.  Are we expecting that people put their classes inside this package so they can make use of it?  Also, this interface isn't exposed anywhere in in the public API surface, so there would be no point in extending or implementing it.

If there is a use case here, maybe that needs some more elaboration?